pkg/actions/db_service.go

```diff
@@ -184,19 +186,23 @@ func Migrate(ctx context.Context, d *pgxpool.Pool, writer io.Writer) error {
 
 	rows, err := d.Query(ctx, "SELECT * FROM actions_runs ORDER BY run_id DESC")
 	if err != nil {
-		return err
+		return kvpg.MigrateErr(err, model.PackageName)
 	}
 	defer rows.Close()
 	rowScanner := pgxscan.NewRowScanner(rows)
 	for rows.Next() {
 		m := new(runResultMigrate)
 		err = rowScanner.Scan(m)
 		if err != nil {
-			return err
+			close(tasksChan)
+			close(runChan)
+			return kvpg.MigrateErr(err, model.PackageName)
```

Comment:
What happens to routines that already pulled from the cahnnel and are currently running? No one awaits them?
I think you may terminate before all routines are finished

Reply:
Correct me if I'm wrong, but since they are waiting on the channel (no jobs actually issued) once we close the channel, these go routines will exit immediately